README.md

```diff
@@ -46,9 +57,10 @@ avr-g++ -O2 -x c++ -isystem /my_path/avr-libstdcpp/include -mmcu=atmega328p test
 
 ```cpp
 #include <array>
+#include <cstddef>
 #include <numeric>
 
-std::array<int, 3U> a { 1, 2, 3 };
+std::array<int, static_cast<std::size_t>(3U)> a { 1, 2, 3 };
```

Comment:
What's wrong with the simple `3U`?

Reply:
> wrong with the simple `3U`?

Almost nothing is _wrong_ with it, that's too strong of a word. I match the exact template parameter type of `size_t` instead of `unsigned` with this change. I admit, this is a lot of code text for a very pedantic change.

Reply:
In my opinion this change make the code less readable.
There are also multiple sections in the C++ Code Guidelines about casts, one of them is [_ES.48: Avoid casts_](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts).

Reply:
I wasn't going to comment on these changes, since my concern was mostly the licensing thing and this kind of stylistic thing isn't my business, but since somebody else has mentioned it :) ...

There is at least one GCC target (H8/300 with the `-mint32` switch) where `unsigned` is wider than `size_t`, but that doesn't apply to avr, and even if it did, `3U` can be implicitly converted to `size_t` without narrowing. So I see no benefit to doing this. The language guarantees that using `3U` is fine (as is using `3`) and this just adds noise for no benefit.

Reply:
Or to put it another way, template parameters are strongly-typed and (with a conforming compiler) will not allow lossy conversions, so whether you write `3` or `3U` or `static_cast<std::size_t>(3u)` there, it's going to be `std::size_t(3u)` in every case. This isn't like calling `printf` where you need to get the types right to avoid implicit conversions that aren't value-preserving.

Reply:
I'm happy removing the casts, but I'll keep the original suffix-`U`. It may be of some clerical interest that GCC warns on `-Wsign-conversion` when implicitly changing sign.

Reply:
> `-Wsign-conversion`
           Warn for implicit conversions that may change the sign of an integer value

The compiler knows for a fact that converting `3` to `size_t` doesn't change the value, so won't warn here. If it *would* change the value, it would be ill-formed:

```
$ ~/gcc/avr/bin/avr-g++  un.cc -Wsign-conversion
un.cc:2:19: error: narrowing conversion of ‘-3’ from ‘int’ to ‘unsigned int’ [-Wnarrowing]
    2 | std::array<int, -3> a;
      |                   ^
```

Either the conversion is value-preserving (and known to be at compile time, so doesn't warn) or it's an narrowing conversion.

If you compile with `-Wno-error=narrowing` then you do get a -Wsign-conversion warning, but it's still ill-formed in this case because you get an impossibly-large array:

```
$ ~/gcc/avr/bin/avr-g++  un.cc -Wsign-conversion -Wno-error=narrowing
un.cc:2:19: warning: narrowing conversion of ‘-3’ from ‘int’ to ‘unsigned int’ [-Wnarrowing]
    2 | std::array<int, -3> a;
      |                   ^
un.cc:2:19: warning: unsigned conversion from ‘int’ to ‘unsigned int’ changes value from ‘-3’ to ‘65533’ [-Wsign-conversion]
In file included from un.cc:1:
/home/jwakely/gcc/avr/avr/include/c++/13.0.1/array: In instantiation of ‘struct std::__array_traits<int, 65533>’:
/home/jwakely/gcc/avr/avr/include/c++/13.0.1/array:109:55:   required from ‘struct std::array<int, 65533>’
un.cc:2:21:   required from here
/home/jwakely/gcc/avr/avr/include/c++/13.0.1/array:55:13: error: size ‘65533’ of array exceeds maximum object size ‘32767’
   55 |       using _Type = _Tp[_Nm];
      |             ^~~~~
```

So using `3U` is no less error-prone, and doesn't avoid any warnings (except in cases that won't compile anyway).

Reply:
Let's continue this discussion at other place/time... I've empirically experienced different aspects than those exhibited by one GCC compiler.